include/readout/NDReadoutTypes.hpp

```diff
@@ -60,11 +54,42 @@ struct PACMAN_MESSAGE_STRUCT
 
   uint64_t get_message_type() const // NOLINT(build/unsigned)
   {
-    dunedaq::dataformats::PACMANFrame translator;
-    return *(translator.get_msg_type((void*)&data)); // NOLINT
-    // return *(reinterpret_cast<const dunedaq::dataformats::PACMANFrame*>(&data)->get_msg_type()); // NOLINT
+    return reinterpret_cast<const dunedaq::dataformats::PACMANFrame*>(&data)->get_msg_header((void *) &data)->type;
   }
+  void inspect_message() const
+  {
+
+    std::cout << "Message timestamp: " << get_timestamp() << std::endl;
+
+    std::cout << "Message Type: " << (char)get_message_type() << std::endl;
+
+    uint16_t numWords = reinterpret_cast<const dunedaq::dataformats::PACMANFrame*>(&data)->get_msg_header((void *) &data)->words;
+
+    std::cout << "Num words in message: " << numWords << std::endl;
+
+    for(unsigned int i = 0; i < numWords; i++)
+    {
 
+      std::cout << "Inspecting word " << i << std::endl;
+
+      dunedaq::dataformats::PACMANFrame::PACMANMessageWord* theWord = reinterpret_cast<const dunedaq::dataformats::PACMANFrame*>(&data)->get_msg_word((void *) &data, i);
+
+      
+      std::cout << "Word type: " << (char)theWord->data_word.type << std::endl;
+      std::cout << "PACMAN I/O Channel: " << (char)theWord->data_word.channel_id << std::endl;
+      std::cout << "Word receipt timestamp: " << theWord->data_word.receipt_timestamp << std::endl;
+      
+      dunedaq::dataformats::PACMANFrame::LArPixPacket* thePacket = &(theWord->data_word.larpix_word);
+
+      std::cout << "Inspecting packet" << std::endl;
+
+      std::cout << "Packet Type: " << thePacket->data_packet.type << std::endl;
+      std::cout << "Packet Chip ID: " << thePacket->data_packet.chipid << std::endl;  
+      std::cout << "Packet Channel ID: " << thePacket->data_packet.channelid << std::endl;
+
+      std::cout << "packet timestamp: " << thePacket->data_packet.timestamp<< std::endl;
+    }
+  }
```

Comment:
Try to use `TLOG` instead of `std::cout`

Reply:
Have replaced with TLOG in latest commit. I also addressed some of the lint warnings. Removing c-style casts may take more significant modifications to the underlying frame, so will review this during a later iteration.